src/backend/distributed/citus.sql

```diff
@@ -191,6 +191,27 @@ COMMENT ON FUNCTION master_create_distributed_table(table_name regclass,
                                                     distribution_method citus.distribution_type)
     IS 'define the table distribution functions';
 
+CREATE FUNCTION master_update_shard_statistics(relation_id regclass, shard_id bigint)
+    RETURNS bigint
+    LANGUAGE C STRICT
+    AS 'MODULE_PATHNAME', $$master_update_shard_statistics$$;
+COMMENT ON FUNCTION master_update_shard_statistics(relation_id regclass, shard_id bigint)
+    IS 'update shard statistics and return the updated shard size';
+
+CREATE FUNCTION is_distributed_table(relation_id regclass)
+    RETURNS boolean
+    LANGUAGE C STRICT
+    AS 'MODULE_PATHNAME', $$is_distributed_table$$;
+COMMENT ON FUNCTION is_distributed_table(relation_id regclass)
+    IS 'check if the given relation is distributed';
+
+CREATE FUNCTION partition_type(relation_id regclass)
```

Comment:
Why not SELECT * FROM master_get_table_metadata? Then you can just get the 3rd column.